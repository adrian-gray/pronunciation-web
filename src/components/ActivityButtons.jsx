import React from 'react'

import ActivityButton from './ActivityButton'

const openActivities = ['how to pronounce', 'movie', 'words', 'phrases',
  'dialogues', 'common words', 'tongue twisters']
const memberActivities = ['minimal pairs', 'find the words',
  'say the sentences', 'odd one out', 'beginner news stories',
  'intermediate news stories', 'words maze']
const pendingActivities = ['hear the words', 'pronunciation journey',
  'sort the words', 'four in a row', 'rhyming words', 'count the sounds',
  'write the minimal pair', 'join to make words', 'listen to the sounds',
  'minimal pairs dominos']

const getStatus = (activity) => {
  if (openActivities.includes(activity)) return 'open'
  if (memberActivities.includes(activity)) return 'member'
  if (pendingActivities.includes(activity)) return 'pending'
  return undefined
}

const ActivityButtons = (props) => {
  const { activityNames } = props

  return activityNames.map(activity => (
    <ActivityButton
      activity={activity}
      key={activity}
      accessStatus={getStatus(activity)}
      {...props}
    />
  ))
}

export default ActivityButtons
