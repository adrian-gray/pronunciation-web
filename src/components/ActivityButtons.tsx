import React from "react";

import ActivityButton from "./ActivityButton";
import {
  openActivities,
  memberActivities,
  pendingActivities
} from "./../data/activityAccess";

const getStatus = (activity: string) => {
  if (openActivities.includes(activity)) return "open";
  if (memberActivities.includes(activity)) return "member";
  if (pendingActivities.includes(activity)) return "pending";
  return undefined;
};

const ActivityButtons = (props: {
  phoneme: string;
  activityNames: string[];
  subscriptionLevel: number;
}) => {
  const { activityNames } = props;
  const buttons = activityNames.map(activity => (
    <ActivityButton
      activity={activity}
      key={activity}
      accessStatus={getStatus(activity)}
      subscriptionLevel={props.subscriptionLevel}
      phoneme={props.phoneme}
      {...props}
    />
  ));

  return <div>{buttons}</div>;
};

export default ActivityButtons;
