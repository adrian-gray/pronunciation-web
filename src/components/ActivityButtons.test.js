import React from "react";
import { BrowserRouter } from "react-router-dom";

import { render, cleanup } from "@testing-library/react";

import ActivityButtons from "./ActivityButtons";

afterEach(cleanup);

test("ActivityButtons", async () => {
  const props = {
    activityNames: [
      "foo",
      "bar",
      "baz",
      "quox",
      "apple",
      "banana",
      "chad",
      "elle"
    ]
  };

  const output = render(
    <BrowserRouter>
      <ActivityButtons {...props} />
    </BrowserRouter>
  );

  expect(output.container).toMatchSnapshot();
});
