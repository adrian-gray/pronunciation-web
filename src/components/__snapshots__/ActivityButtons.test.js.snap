// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityButtons 1`] = `
<div>
  <a
    class="MuiChip-root makeStyles-chip-1 undefined MuiChip-clickable"
    href="/sound/undefined/foo"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label"
    >
      Foo
    </span>
  </a>
  <a
    class="MuiChip-root makeStyles-chip-1 undefined MuiChip-clickable"
    href="/sound/undefined/bar"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label"
    >
      Bar
    </span>
  </a>
  <a
    class="MuiChip-root makeStyles-chip-1 undefined MuiChip-clickable"
    href="/sound/undefined/baz"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label"
    >
      Baz
    </span>
  </a>
  <a
    class="MuiChip-root makeStyles-chip-1 undefined MuiChip-clickable"
    href="/sound/undefined/quox"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label"
    >
      Quox
    </span>
  </a>
  <a
    class="MuiChip-root makeStyles-chip-1 undefined MuiChip-clickable"
    href="/sound/undefined/apple"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label"
    >
      Apple
    </span>
  </a>
  <a
    class="MuiChip-root makeStyles-chip-1 undefined MuiChip-clickable"
    href="/sound/undefined/banana"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label"
    >
      Banana
    </span>
  </a>
  <a
    class="MuiChip-root makeStyles-chip-1 undefined MuiChip-clickable"
    href="/sound/undefined/chad"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label"
    >
      Chad
    </span>
  </a>
  <a
    class="MuiChip-root makeStyles-chip-1 undefined MuiChip-clickable"
    href="/sound/undefined/elle"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiChip-label"
    >
      Elle
    </span>
  </a>
</div>
`;
